File: src/quark/designdocs/trades.design/views/chart/map.js
function(doc) {
  if (doc.time) {
	emit([(doc.time/604800|0),(doc.time/86400|0)%7,(doc.time/14400|0)%6,(doc.time/3600|0)%4, (doc.time/900|0)%4,(doc.time/300|0)%3, (doc.time/60|0)%5], [doc.price,doc.size]);
}}

File: src/quark/designdocs/trades.design/views/chart/map.cpp
#include <couchcpp/api.h>

void mapdoc(Document doc) {
	  if (doc.getID().substr(0,2) == "t.") {
		  auto tm = doc["time"].getUInt();
		  emit({
			  tm/604800,
			  (tm/86400)%7,
			  (tm/14400)%6,
			  (tm/3600)%4,
			  (tm/900)%4,
			  (tm/300)%3,
			  (tm/60)%5
		  }, {doc["price"],doc["size"],doc["index"]});
	}
}
